src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,123 @@
+import classNames from 'classnames';
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todo: Todo;
+  onDelete?: (todoId: number) => Promise<void>;
+  onUpdate?: (id: number, body: Partial<Omit<Todo, 'id'>>) => Promise<void>;
+  isLoading?: boolean;
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onDelete = () => {},
+  onUpdate = () => {},
+  isLoading = true,
+}) => {
+  const {
+    id,
+    title,
+    completed,
+  } = todo;
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [inputValue, setInputValue] = useState(title);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current !== null) {
+      inputRef.current.focus();
+    }
+  }, [isEditing]);
+
+  const handleSubmit = async (
+    event: React.FormEvent<HTMLFormElement>,
+  ) => {
+    event.preventDefault();
+
+    const trimmedInputValue = inputValue.trim();
+
+    if (trimmedInputValue && trimmedInputValue !== title) {
+      await onUpdate(id, { title: inputValue });
+    }
+
+    if (!inputValue) {
```

Comment:
It's better to check for `trimmedInputValue` before calling `onDelete` instead of checking for `!inputValue`. This prevents the deletion of tasks with whitespace.